### PR TITLE
migrated app router

### PR DESCRIPTION
**Pull request by Lada496**
### Feature Description
- convert `pages/_app.tsx` and `pages/_document.tsx` to `app/layout.tsx` (ref:[Migrating _document.js and _app.js](https://nextjs.org/docs/pages/building-your-application/upgrading/app-router-migration#migrating-_documentjs-and-_appjs))
- onvert components/SEO.tsx feature to each `metadata`(ref: [Step 3: Migrating next/head](https://nextjs.org/docs/pages/building-your-application/upgrading/app-router-migration#step-3-migrating-nexthead))
- Created client components for components/providers that require client side rendering
- Configured tRPC with `createTRPCReact` instead of `createTRPCNext` as I concluded this is better option for now  (refs: https://github.com/vercel/next.js/issues/46859, https://codevoweb.com/setup-trpc-server-and-client-in-nextjs-13-app-directory/, https://github.com/trpc/next-13/blob/3992fb41ec4b1134d596d94007fdea51453445dd/client/trpcClient.tsx, )

### Related Issue
N/A

### Something you want to mention
- I decided to keep `utils/api.ts` for now since there're still some possibilities that we want to get back `createTRPCNext`
- We need to create client components for each Mantine component to make the most of server-side rendering. Otherwise, we have to turn most of the pages to "use client" mode, which prohibits us to use matadata ([ref](https://nextjs.org/docs/app/api-reference/functions/generate-metadata))

### Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
-const Home: NextPage = () => {
+export const metadata: Metadata = {
+  title: "Intro | app name",
+  description: "Generated by create-t3-app",
+};
+const HomePage = () => {
   return (
     <>
-      <SEO title="Intro" />
       <main className="flex h-screen flex-col items-center justify-center">
```
Comment (devjoylee) [inline src/app/page.tsx:8]: I wrote this line temporarily and I was wondering if we use both Tailwind and Mantine together, or can we style it only with Mantine?

File: src/clients/components/Button.tsx
Comment (devjoylee) [inline src/clients/components/Button.tsx:1]: I like the idea of making separate components for Mantine. It looks much better now!

**Review by devjoylee — COMMENTED** [review]
Thank you for setting up and the references. 
I fixed the environmental variables and now it's successfully deployed on Netlify. 
However, the warning mark for this PR won't be fixed unless you re-open this PR.. would it be okay?

**Comment by Lada496** [comment]
@devjoylee Thank you for setting up netlify!
I updated some code to and now I got ✅ :)

File: src/app/page.tsx (hunk 1 above)
Comment (Lada496) [reply src/app/page.tsx:8]: In my opinion, we can mainly use Mantine as much as possible. 
We can set up some sharing layout for both pages later with Mantine.

File: src/clients/components/Button.tsx
Comment (Lada496) [reply src/clients/components/Button.tsx:1]: I think we can set dark/light stylings here later

**Pull request merged**